cmake/ecbuild_check_os.cmake

```diff
@@ -372,6 +372,45 @@ if( ${CMAKE_SYSTEM_NAME} MATCHES "CYGWIN" )
 
 endif()
 
+### Windows
+
+if( WIN32 )
+
+  set( EC_OS_NAME "windows" )
+
+  # Suppress compliler warnings - see ECC-850
+  # Suppress warnings about using 'insecure' functions. Fixing this would require changes all over
+  # the codebase which would damage portability.
+  ecbuild_add_c_flags("/D_CRT_SECURE_NO_WARNINGS")
+  ecbuild_add_cxx_flags("/D_CRT_SECURE_NO_WARNINGS")
+  # Suppress warnings about using well-known C functions.
+  ecbuild_add_c_flags("/D_CRT_NONSTDC_NO_DEPRECATE")
```

Comment:
Please consider very carefully which flags to put here. The principle is to avoid at all costs setting flags here.
Setting flags here, implies that any project that uses ecBuild to build in windows will inherit them with no choice but to use them.
My analysis is that most of these flags are project specific and therefore belong on the CMake code of the projects that use ecBuild.

Reply:
Fair point. I've moved the flag setting to ecCodes and Magics.